stmhal/main.c

```diff
@@ -68,6 +68,7 @@ static FATFS fatfs1;
 #endif
 
 void flash_error(int n) {
+#if MICROPY_HW_HAS_LED
```

Comment:
ok - I went back and forth a few times. I thought about just addng an extra define for LED1 and using that, but then having the HW_HAS_LED seemed more consistent.

So I'd like to get some feedback from you about how you want to deal with it.

I see 3 ways:
1 - Use MICROPY_HW_HAS_LED
2 - Use MICROPY_HW_LED1

With either of those, we can choose to use the dummy functions, or not.